app/src/main/java/com/teamagam/gimelgimel/app/map/actions/timeplay/TimeplayViewModel.java

```diff
@@ -200,4 +252,61 @@ public void clearMap() {
       mDisplayed.clear();
     }
   }
+
+  public class TextTimeDisplayer implements RadialTimePickerDialogFragment.OnTimeSetListener {
+
+    private boolean mIsInEditOfStartDate;
+    private Calendar mResultCalender;
+
+    public TextTimeDisplayer(boolean isInEditOfStartDate) {
+      mIsInEditOfStartDate = isInEditOfStartDate;
+    }
+
+    @Override
+    public void onTimeSet(RadialTimePickerDialogFragment dialog, int hour, int minute) {
+      updateResultDateWithTime(hour, minute);
+      if (mIsInEditOfStartDate) {
+        updateStartDate();
+      } else {
+        updateEndDate();
+      }
+    }
+
+    public void setResultCalender(Calendar resultCalender) {
+      mResultCalender = resultCalender;
+    }
+
+    private void updateEndDate() {
```

Comment:
I wouldn't expect updateEndState to do any validation

Reply:
done